Update the getting started guide to use the FlyBy API

rkoron007 opened the pull request:
The Apollo Client [Getting started](https://www.apollographql.com/docs/react/get-started) article currently relies on a Codesandbox instance that queries the Coinbase API. We decided that it'd be better to better to update this article to use an API that Apollo manages and updates (I ended up going with the [FlyBy API ](https://odyssey-flyby.netlify.app/) from [Voyage I](https://www.apollographql.com/tutorials/voyage-part1/intro-to-federation)).

rkoron007 commented:
@jpvajda Wasn't sure if you wanted to review this too— but wanted to let you know we were updating the "Getting Started" guide! (I also added a note about downgrading `graphql` for those running into this problem: https://github.com/apollographql/apollo-client/issues/9659 )

StephenBarlow commented on docs/source/get-started.mdx:
```diff
@@ -2,57 +2,57 @@
+Hello! 👋 This short tutorial gets you up and running with Apollo Client.
 
-> For a more complete introduction to the entire Apollo platform, [check out **Odyssey**, Apollo's new learning platform](https://odyssey.apollographql.com?utm_source=apollo_docs&utm_medium=referral).
+> For an introduction to the entire Apollo platform, [check out **Odyssey**, Apollo's new learning platform](https://odyssey.apollographql.com?utm_source=apollo_docs&utm_medium=referral).
```
```suggestion
> For an introduction to the entire Apollo platform, [check out **Odyssey**, Apollo's interactive learning platform](https://www.apollographql.com/tutorials/?utm_source=apollo_docs&utm_medium=referral).
```

StephenBarlow replied:
Might as well cut "new" at this point 😄

StephenBarlow commented on docs/source/get-started.mdx:
````diff
@@ -2,57 +2,57 @@
+```
 
-## 2. Initialize `ApolloClient`
+> If you are using a React sandbox and running into issues, try downgrading your `graphql` package. You can use `npm install graphql@15.8.0` to downgrade.
````
Is this specifically about an issue within Codesandbox, or can it also occur locally? We might need to make this more general if both

StephenBarlow commented on docs/source/get-started.mdx:
```diff
@@ -2,57 +2,57 @@
+## Step 1: Setup
 
-To walk through this tutorial, we recommend you either:
+Before starting this tutorial, we recommend you either:
```
```suggestion
To start this tutorial, do one of the following:
```

StephenBarlow commented on docs/source/get-started.mdx:
```diff
@@ -2,57 +2,57 @@
+> If you are using a React sandbox and running into issues, try downgrading your `graphql` package. You can use `npm install graphql@15.8.0` to downgrade.
 
-Now that we have the dependencies we need, let's initialize an `ApolloClient` instance.
+We'll use the [FlyBy API](https://flyby-gateway.herokuapp.com/) from the [Voyage I: Federation from Day One](https://www.apollographql.com/tutorials/voyage-part1/) Odyssey course as our GraphQL server for this example. The Flyby API provides a list of intergalactic locations and details about those locations.
```
```suggestion
Our example application will use the [FlyBy GraphQL API](https://flyby-gateway.herokuapp.com/) from the [Voyage tutorial series](https://www.apollographql.com/tutorials/voyage-part1/) in Apollo Odyssey. This API provides a list of intergalactic travel locations and details about those locations 👽
```

StephenBarlow replied:
(alien emoji very optional 😄 )

StephenBarlow commented on docs/source/get-started.mdx:
```diff
@@ -2,57 +2,57 @@
+
+With our dependencies set up, we can now initialize an `ApolloClient` instance.
+
+In `index.js`, let's first import the things we need from `@apollo/client`:
```
```suggestion
In `index.js`, let's first import the symbols we need from `@apollo/client`:
```

StephenBarlow replied:
Avoid "things" wherever possible

StephenBarlow commented on docs/source/get-started.mdx:
```diff
@@ -2,57 +2,57 @@
 
-* `uri` specifies the URL of our GraphQL server.
-* `cache` is an instance of `InMemoryCache`, which Apollo Client uses to cache query results after fetching them.
+More on the `uri` and `cache` fields below:
```
```suggestion
```

StephenBarlow replied:
Despite flying in the face of the style guide, I personally don't think this particular list requires an intro sentence because it's directly addressing the immediately preceding codeblock. But feel free to keep if you disagree!

StephenBarlow commented on docs/source/get-started.mdx:
```diff
@@ -110,69 +107,76 @@ function App() {
-      currency
-      rate
+const GET_LOCATIONS = gql`
+  query getLocations {
```
```suggestion
  query GetLocations {
```

StephenBarlow commented on docs/source/get-started.mdx:
```diff
@@ -110,69 +107,76 @@ function App() {
 
-* Apollo Client tracks a query's error and loading state for you, which are reflected in the `loading` and `error` properties.
-* When the result of your query comes back, it's attached to the `data` property.
+- Apollo Client tracks a query's error and loading states for you, reflected in the `loading` and `error` properties.
```
```suggestion
- Apollo Client automatically tracks a query's loading and error states, which are reflected in the `loading` and `error` properties.
```

StephenBarlow replied:
Agree this sentence needed a tweak, realized we should probably list "loading" and "error" in the same order in both places

StephenBarlow commented on docs/source/get-started.mdx:
````diff
@@ -110,69 +107,76 @@ function App() {
 ```
 
-When your app reloads, you should briefly see a loading indicator, followed by a list of exchange rates! If you don't, you can compare your code against the [completed app on CodeSandbox](https://codesandbox.io/s/get-started-coinbase-client-73r10).
+When your app reloads, you should briefly see a loading indicator, followed by a list of locations and details about those locations! If you don't, you can compare your code against the [completed app on CodeSandbox](https://codesandbox.io/s/get-started-apollo-client-demo-bert8s?file=/src/index.js).
````
```suggestion
When your app reloads, you should briefly see a loading indicator, followed by a list of locations and details about those locations! If you don't, you can compare your code against the [completed app on CodeSandbox](https://codesandbox.io/s/get-started-apollo-client-demo-bert8s).
```

StephenBarlow commented on docs/source/get-started.mdx:
```diff
@@ -64,42 +64,39 @@ In the same `index.js` file, call `client.query()` with the query string (wrappe
-      query GetRates {
-        rates(currency: "USD") {
-          currency
+      query getLocations {
```
```suggestion
      query GetLocations {
```

StephenBarlow replied:
PascalCase for operation names

StephenBarlow reviewed (changes requested):
Looking good! Finished a feedback pass

jpvajda reviewed (approved):
This looks great from my perspective, thanks for improving this guide!

StephenBarlow commented on docs/source/get-started.mdx:
```diff
@@ -64,42 +63,39 @@ In the same `index.js` file, call `client.query()` with the query string (wrappe
+  cache: new InMemoryCache(),
 });
 
 function App() {
```
In both my CRA app and the default codesandbox React template, `App` is pulled in from `App.js` as opposed to defined directly in `index.js`. I reckon we should conform to the same pattern here.

StephenBarlow replied:
In fact yes, let's do that and move all of the actual query logic and the component structure into `App.js`

rkoron007 commented:
As per our conversation @StephenBarlow, I updated this tutorial to directly use the `App.js` file instead of putting everything in `index.js`. I also updated the CodeSandbox to match!

StephenBarlow commented on docs/source/get-started.mdx:
````diff
@@ -64,115 +62,128 @@ In the same `index.js` file, call `client.query()` with the query string (wrappe
-  useQuery,
-  gql
-} from "@apollo/client";
+```jsx title="index.js" {22-24}
````
```suggestion
```jsx title="index.js" {15-17}
```

StephenBarlow replied:
Highlighted lines got moved a bit

StephenBarlow commented on docs/source/get-started.mdx:
````diff
@@ -2,57 +2,55 @@
+npm install @apollo/client graphql
+```
 
+> If you are using a React sandbox from CodeSandbox and running into issues, try downgrading your `graphql` package. You can use `npm install graphql@15.8.0` to downgrade.
````
```suggestion
> If you're using a React sandbox from CodeSandbox and you encounter a `TypeError`, try downgrading the version of the `graphql` package to `15.8.0` in the Dependencies panel. If you encounter a _different_ error after downgrading, refresh the page.
```

StephenBarlow reviewed (approved):
Woo thanks for incorporating all these pesky changes! Just a last couple things and then merge whenever's good! 🎉
